src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/Event.scala

```diff
@@ -167,14 +171,48 @@ case class Event(app_id:                   Option[String],
   }
 }
 
+
 object Event {
-  implicit val jsonEncoder: Encoder[Event] = deriveEncoder[Event]
+  case class Lossy(event: Event) extends AnyVal
+  case class Lossless(event: Event) extends AnyVal
 
+  implicit val jsonEncoder: Encoder[Event] = deriveEncoder[Event]
+  implicit val losslessEncoder: Encoder[Lossless] = Encoder.instance { lossless: Lossless => lossless.event.asJson }
+  implicit val lossyEncoder: Encoder[Lossy] = Encoder.instance { lossy: Lossy =>
+    val event = lossy.event
+    var json = event.asJson.asObject.get
+      .remove("contexts")
+      .remove("unstruct_event")
+      .remove("derived_contexts")
+    event.contexts.data.groupBy(x => (x.schema.vendor, x.schema.name, x.schema.format, x.schema.version.model)).foreach {
+      case ((vendor, name, format, model), data) =>
+        json = json.add(transformSchema("contexts", vendor, name, model), data.map{selfdesc => selfdesc.data}.asJson)
+    }
+    event.unstruct_event.data.foreach{
+      case SelfDescribingData(schema, data) =>
+        json = json.add(transformSchema("unstruct_event", schema.vendor, schema.name, schema.version.model), data)
+    }
+    event.derived_contexts.data.groupBy(x => (x.schema.vendor, x.schema.name, x.schema.format, x.schema.version.model)).foreach {
+      case ((vendor, name, format, model), data) =>
+        json = json.add(transformSchema("contexts", vendor, name, model), data.map{selfdesc => selfdesc.data}.asJson)
+    }
+    json.asJson
+  }
   private val parser: Parser[Event] = Parser.deriveFor[Event].get
 
   def parse(line: String): RowDecodeResult[Event] =
     parser.parse(line)
 
-  def asJson(event: Event): Json =
-    event.asJson
+  def asJson(event: Event, lossy: Boolean = false): Json =
+    if (lossy) {
+      Lossy(event).asJson
```

Comment:
I still don't see a point of these wrapper classes *if they don't bring any automation*